packages/itwin/reports-config-widget/src/widget/components/ExtractionStatus.tsx

```diff
@@ -0,0 +1,150 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { ProgressRadial } from "@itwin/itwinui-react";
+import * as React from "react";
+import { useEffect, useState } from "react";
+import "./ExtractionStatus.scss";
+import {
+  SvgStatusError,
+  SvgStatusPending,
+  SvgStatusPendingHollow,
+  SvgStatusSuccess,
+} from "@itwin/itwinui-icons-color-react";
+import { ReportsConfigWidget } from "../../ReportsConfigWidget";
+
+export enum ExtractionStates {
+  None,
+  Starting,
+  FetchingUpdate,
+  Queued,
+  Running,
+  Succeeded,
+  Failed,
+}
+interface ExtractionStatusProps {
+  state: ExtractionStates;
+  setExtractionState?: React.Dispatch<React.SetStateAction<ExtractionStates>>;
+  children?: React.ReactNode;
+}
+
+export const ExtractionStatus = ({
+  state,
+  children,
+  setExtractionState,
+}: ExtractionStatusProps) => {
+  const [fadeOut, setFadeOut] = useState<boolean>(false);
+
+  const onAnimationEnd = () => {
+    if (setExtractionState) {
+      setExtractionState(ExtractionStates.None);
+      setFadeOut(false);
+    }
+  };
+
+  useEffect(() => {
+    let timer: number;
+    switch (state) {
+      case ExtractionStates.Succeeded:
+      case ExtractionStates.Failed:
+        timer = window.setTimeout(() => setFadeOut(true), 5000);
+    }
+    return () => clearTimeout(timer);
+  }, [state, setExtractionState]);
+
+  switch (state) {
+    case ExtractionStates.Starting:
+      return (
+        <div
```

Comment:
Maybe the contents of each `case` could be moved into separate files? That would make the `ExtractionStatus` component smaller and more readable.

Move all of these components into a subdirectory.

Reply:
Each `ExtractionState` has been moved into a separate file in `ExtractionStates` directory